### DISCUSSION TITLE
Message Delivery (XEP-0184, XEP-0333)

### DISCUSSION THREAD

**Discussion by AtifShahzed** [vitalyster/SharpXMPP#117] [Ideas]
Hi,

I am using this library and pretty happy. Now I am looking to get information about the message delivery.
Is there any way I can track the message delivery information.

Thanks

**Comment by vitalyster** [comment]
Delivery information properties are not implemented as convenient fields of XMPPMessage, but you can inspect them with XML to LINQ queries and find corresponding payloads described in XEP-0184 or XEP-0333, depending on which delivery information XEPs other clients are support. You may also need to add required features such as `urn:xmpp:receipts` to your client discovery information to start receiving delivery requests and receipts
